### PR TITLE
split mltrain

### PR DESCRIPTION
**Pull request by vbrunn**
split mltrain into traindata (enabling extration of preliminary tree map in rast and vect format) and mltrain (doing the random forest training)

### CODE REVIEW

**Comment by griembauer** [comment]
Todo: Update top-level Readme - all addons are listed and summarized there

File: README.md [hunk 1]
```diff
@@ -39,8 +39,9 @@ Repo for code and script transfer between mundialis and RVR - GRASS GIS addons:
 - **m.analyse.trees**
 
   - **r.trees.peaks** assigns pixels to nearest peak (tree crown).
-  - **r.trees.mltrain** generates training data for a machine learning (ML)
-    model to detect trees and trains the model with these training data.
+  - **r.trees.peaks** generates training data for a machine learning (ML)
```
Comment (anikaweinmann) [inline README.md:42]:
```suggestion
  - **r.trees.traindata** generates training data for a machine learning (ML)
```

**Review by anikaweinmann — APPROVED** [review]

File: grass-gis-addons/m.analyse.trees/r.trees.traindata/r.trees.traindata.py [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+# % type: double
+# % required: yes
+# % label: Area size threshold for potential trees
+# % answer: 5
+# % guisection: Parameters
+# %end
+
+# %option G_OPT_R_OUTPUT
+# % key: traindata_r
+# % label: primary tree map as raster
```
Comment (griembauer) [inline grass-gis-addons/m.analyse.trees/r.trees.traindata/r.trees.traindata.py:146]:
```suggestion
# % label: Preliminary tree map as raster
```

File: grass-gis-addons/m.analyse.trees/r.trees.traindata/r.trees.traindata.py [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+# % description: Primary tree map in raster format to pass to mltrain
+# % required: no
+# % guisection: Output
+# %end
+
+# %option G_OPT_V_OUTPUT
+# % key: traindata_v
+# % label: primary tree map as vector
+# % description: Primary tree map in vector format to export and edit
```
Comment (griembauer) [inline grass-gis-addons/m.analyse.trees/r.trees.traindata/r.trees.traindata.py:155]:
```suggestion
# % description: Preliminary tree map in vector format to export and edit
```

**Review by griembauer — APPROVED** [review]
Thanks! Just a very small naming suggestion

**Pull request merged**